src/core/net/dns_headers.cpp

```diff
@@ -207,27 +207,31 @@ otError Name::ParseName(const Message &aMessage, uint16_t &aOffset)
     {
         error = iterator.GetNextLabel();
 
-        VerifyOrExit((error == OT_ERROR_NONE) || (error == OT_ERROR_NOT_FOUND));
-
-        if (iterator.IsEndOffsetSet())
+        switch (error)
         {
+        case OT_ERROR_NONE:
+            break;
```

Comment:
should we exit in the case `error == OT_ERROR_NONE` same as the original code did?

Reply:
BTW, I think the method name should be renamed to `SkipName` to reflect its actual behavior.

Reply:
I think calling `ExitNow` causes extra instruction (`jmp`) and brings one more line of code.

Reply:
> should we exit in the case error == OT_ERROR_NONE same as the original code did?

The original code did not exit on `error == OT_ERROR_NONE`. And it would not be not correct behavior and not what we want.

The behavior we want is to keep reading labels till we can get to the end of name (or in original code, till we find the end offset).

The original code was doing `VerifyOrExit((error == OT_ERROR_NONE) || (error == OT_ERROR_NOT_FOUND));` which exits if error is anything other than these two values. This is equivlaent to `default` in the new `switch(aError)` statement.

> BTW, I think the method name should be renamed to `SkipName` to reflect its actual behavior.

Skipping over the read info is somthing that all similar API already do (e.g. `ReadName()` also skips over the name). I prefer `ParseName` to be clear that we are parsing the name (i.e. verifying it is correctly formatted/encoded in the message).

Reply:
thanks for explanation.